### PR TITLE
ref(Dynamic-Sampling): Add drag and drop feature

### PR DESCRIPTION
**Pull request by priscilawebdev**
- Introduce the package [React-beautiful-dnd](https://github.com/atlassian/react-beautiful-dnd) to deal with drag-and-drop. 

   **Minified:** 106.5kB
   **Minified + Gzipped:** 31.2kB

- Add Placeholder component

ps: A common drag-and-drop component based on this library is planned 

This PR introduces the feature below:

filters-and-sampling-example

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/index.tsx [hunk 1]
```diff
@@ -1,83 +1,205 @@
...
-          onEditRule={onEditRule(rule)}
-          onDeleteRule={onDeleteRule(rule)}
-          disabled={disabled}
-        />
-      ))}
-    </StyledPanelTable>
-  );
-}
+type State = {
+  placeholderProps?: {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/index.tsx:32]: I think we should reconsider if we need a placeholder. The new library does a good job of providing feedback on where an item will end up, and we could use background colors to create contrast between the gap and other items.

File: src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/rule/index.tsx [hunk 2]
```diff
@@ -39,22 +59,37 @@ function Rule({rule, onEditRule, onDeleteRule, disabled}: Props) {
...
+  :not(:last-child) {
+    border-bottom: 1px solid ${p => p.theme.border};
+  }
+  ${p => layout(p.theme)}
+  ${p =>
+    p.isDragging &&
+    `
+      background: ${p.theme.background};
+      box-shadow: 0 0 15px rgba(0, 0, 0, 0.15);
+      opacity: 0.8;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/rule/index.tsx:80]: I wouldn't make the dragging item translucent. It is the current item that the user is interacting with and should feel solid in my opinion.

File: src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/index.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/index.tsx:32]: yes, that is a difficult decision. I like the dashed placeholder, but I agree with you about the complexity that it brings

File: src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/rule/index.tsx (hunk 2 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/rules/rule/index.tsx:80]: I will update that 😉

**Comment by priscilawebdev** [comment]
closing this PR in favor of https://github.com/getsentry/sentry/pull/23497

**Pull request closed without merging**